File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/05_While_Loop_Exercise/01_Old_Books.js
function oldBooks(input) {
    let exitMessage = "No More Books";

    let booksChecked = 0;
    let targetBook = input[0];

    let index = 1;  // index; i;
    let currentBook = input[index];
    let isFound = false;


    while (currentBook != exitMessage) {
        booksChecked++;  // += 1;

        if (currentBook == targetBook) {
            // console.log(`You checked ${booksChecked - 1} books and found it.`);
            isFound = true;
            break;
        }


        index++;
        currentBook = input[index];
    }

    if (isFound === false) {
        console.log("The book you search is not here!");
        console.log(`You checked ${booksChecked} books.`);
    } else {
            console.log(`You checked ${booksChecked - 1} books and found it.`);
    }

}


// oldBooks([
//     "Troy",
//     "Stronger",
//     "Life Style",
//     "Troy",
// ]);

// oldBooks(["The Spot",
//     "Hunger Games",
//     "Harry Potter",
//     "Torronto",
//     "Spotify",
//     "No More Books"]);

// function oldBooks(input) {
//     let exitMessage = "No More Books";

//     let booksChecked = 0;
//     let targetBook = input[0];

//     let currentIndex = 1;  // index; i;
//     while (true) {
//         let currentBook = input[currentIndex];  // 1st iteration => "input[1]"
//         currentIndex++;  // 1st iteration => 2; 2nd iteration => 3;
//         booksChecked++;  // += 1;

//         if (currentBook == exitMessage) {  // true -> if library is done;
//             console.log("The book you search is not here!");
//             console.log(`You checked ${booksChecked - 1} books.`);
//             break;
//         }

//         if (currentBook == targetBook) {
//             console.log(`You checked ${booksChecked - 1} books and found it.`);
//             break;
//         }

//         // currentIndex++;
//     }

// // }

File: Python_Courses/Python_Basics/Preparation/05_While_Loop_Exercise/01_Old_Books.py
# # Solution 1
# checked_books = 0
# target_book = input()
#
#
# command = input()
# while command != "No More Books":
#     current_book = command
#     if current_book == target_book:
#         break
#
#     checked_books += 1
#     command = input()
#
# if command != "No More Books":
#     print(f"You checked {checked_books} books and found it.")
# else:
#     print("The book you search is not here!")
#     print(f"You checked {checked_books} books.")
#
# # Solution 2
# NO_MORE_BOOKS_COMMAND = "No More Books"
#
# checked_books = 0
# target_book = input()
#
# command = input()
# while not command == NO_MORE_BOOKS_COMMAND:
#     current_book = command
#     if current_book == target_book:
#         print(f"You checked {checked_books} books and found it.")
#         break
#
#     checked_books += 1
#     command = input()
# else:
#     print("The book you search is not here!")
#     print(f"You checked {checked_books} books.")
#

# Solution 3
NO_MORE_BOOKS_COMMAND = "No More Books"

checked_books = 0
target_book = input()

is_book_found = False
while True:
    command = input()
    if command == NO_MORE_BOOKS_COMMAND:
        break

    current_book = command
    if current_book == target_book:
        is_book_found = True
        break

    checked_books += 1


if is_book_found:
    print(f"You checked {checked_books} books and found it.")
else:
    print("The book you search is not here!")
    print(f"You checked {checked_books} books.")
